src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceBeforeRecordHeader.java

```diff
@@ -0,0 +1,12 @@
+/*
+GenericWhitespace
+
+
+*/
+
+//non-compiled with javac: Compilable with Java16
+package com.puppycrawl.tools.checkstyle.checks.whitespace.genericwhitespace;
+
+public class InputGenericWhitespaceBeforeRecordHeader {
+    record Session<T>() {}
```

Comment:
**Disclaimer:** Contains some off-topic observations.

The purpose of this input and the corresponding test is to kill the mutation which survived after the changes in Check.
Surviving mutation  was replacing `charAfter == '('` with `false`  in the following line:
https://github.com/checkstyle/checkstyle/blob/9842608983cf00c1927d5e5bfcffdb05f4936afb/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java#L347 <hr>
#### Why this didn't survived before:
Before the fix, check didn't identified all the cases where generic was present before constructor invocation, which meant some `CTOR` cases where `>` was succeeded with `(` weren't filtered through the following `if` statement:
https://github.com/checkstyle/checkstyle/blob/9842608983cf00c1927d5e5bfcffdb05f4936afb/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java#L222
Such cases killed the above mutation.
Due to the fix, all the possible existing cases where `>` could be succeeded by `(` were filtered through the if statement which made the mutant survive.

So after a bit of mutant hunting to find a valid input where `>` could be succeeded by `(` but wasn't before Method or Ctor, found out that such valid input exist for generic before record header, which was able to kill the mutant.

As of now, we don't have clear definition in the documentation on how to handle whitespace for generic that is before a record header, current check does not count space between `>` and `(` of record header as a violation, will open an Issue shortly for discussing if it should, based on the decision of the discussion, will update the Check accordingly and strengthen the test inputs for the records in a separate PR.

Funny thing is that if there is supposed to be violation for the case above, this will revive the mutant, as we will now be filtering the killing input in the if statement(this in my knowledge makes for all possible cases where generic can be preceded by `(`), which will mean that the affected operand could be safely removed if no regressions are found and no one is aware of such input.